File: library/src/main/java/mopel/io/library/TadaAnimator.java
package mopel.io.library;

import android.animation.ObjectAnimator;
import android.view.View;

/**
 * Author: mopel
 * Date : 2017/4/23
 * cssAnimation: link https://github.com/daneden/animate.css/blob/master/source/attention_seekers/tada.css
 *
 */

public class TadaAnimator extends BaseAnimator{
  @Override protected void prepare(View target) {
    getAnimator().playTogether(
        ObjectAnimator.ofFloat(target, "scaleX", 1, 0.9f, 0.9f, 1.1f, 1.1f, 1.1f, 1.1f, 1.1f, 1.1f, 1),
        ObjectAnimator.ofFloat(target, "scaleY", 1, 0.9f, 0.9f, 1.1f, 1.1f, 1.1f, 1.1f, 1.1f, 1.1f, 1),
        ObjectAnimator.ofFloat(target, "rotation", 0, -3, -3, 3, -3, 3, -3, 3, -3, 0)
    );
  }
}

File: lib-kotlin/src/main/java/mopel/io/lib_kotlin/TadaAnimator.kt
package mopel.io.lib_kotlin

import android.animation.ObjectAnimator
import android.view.View

/**
 * Author: mopel
 * Date : 2017/4/23
 * cssAnimation: link https://github.com/daneden/animate.css/blob/master/source/attention_seekers/tada.css

 */

class TadaAnimator : BaseAnimator() {
    override fun prepare(target: View) {
        animator.playTogether(
                ObjectAnimator.ofFloat(target, "scaleX", 1f, 0.9f, 0.9f, 1.1f, 1.1f, 1.1f, 1.1f, 1.1f, 1.1f, 1f),
                ObjectAnimator.ofFloat(target, "scaleY", 1f, 0.9f, 0.9f, 1.1f, 1.1f, 1.1f, 1.1f, 1.1f, 1.1f, 1f),
                ObjectAnimator.ofFloat(target, "rotation", 0f, -3f, -3f, 3f, -3f, 3f, -3f, 3f, -3f, 0f)
        )
    }
}
